main.cpp

```diff
@@ -231,214 +333,320 @@ struct BasketballCourt
+
     void advertiseCompanyProductsOnSocialMedia();
-    //    2) create the company logo
     void createCompanyLogo();
-    //    3) pay the employees
-    float payTheEmployees(float employeeSalary, int numberOfEmployees); //returns the amount of money paid to the employes
-
+    float payTheEmployees(float employeeSalary, int numberOfEmployees);
 };
 
-/*
-Thing 10) audiovisual company
+void FinanceAndAdvertising::advertiseCompanyProductsOnSocialMedia()
+{
+    std::string mainPlatform = "platform";
+    int desiredViews = 19283;
+    std::cout << "advertise on" << mainPlatform << "until you reach" << desiredViews << "views";
+}
 
-*/
+void FinanceAndAdvertising::createCompanyLogo()
+{
+    std::string wordsOnLogo = "audiovisual";
+    std::cout << "create a logo based on" << wordsOnLogo;
+}
+
+float FinanceAndAdvertising::payTheEmployees(float employeesSalary, int totalNumberOfEmployees)
+{
+    float employeesFloat;
+    employeesFloat = float(totalNumberOfEmployees);
+    float amountPaid = employeesSalary * employeesFloat;
+    return amountPaid;
+}
  
-struct audiovisualCompany
+struct AudiovisualCompany
 {
-    //5 properties:
-    //    1) audio department
     AudioDepartment audioDepartment;
-    //    2) visual department
     VisualDepartment visualDepartment;
-    //    3) sales
     Sales sales;
-    //    4) RnD department
     RndDevelopment rndDevelopment;
-    //    5) finance and advertising
     FinanceAndAdvertising financeAndAdvertising;
-    //3 things it can do:
-    //    1) create audiovisual content
+
     void createAudiovisualContent();
-    //    2) develop new audiovisual software
     void developNewAudiovisualSoftware();
-    //    3) sell products to customers
-    float sellProductsToCustomers(float softwarePrice, int softwarePiecesSold, int numberOfSubscribersToCloudSolutions, float subscriptionPrice);  //returns the total earnings  
-
+    float sellProductsToCustomers(float softwarePrice, int softwarePiecesSold, int numberOfSubscribersToCloudSolutions, float subscriptionPrice);
 };
 
+void AudiovisualCompany::createAudiovisualContent()
+{
+    int numberOfUpcomingProductions = 8;
+    std::cout << "develop content for" << numberOfUpcomingProductions << "productions";
+}
 
+void AudiovisualCompany::developNewAudiovisualSoftware()
+{
+    int numberOfWorkingEmployeesOnProject = 12;
+    int numberOfTasksToBeDone = 92;
+    std::string projectManager = "Tom";
+    std::cout << "employ" << projectManager << "to allocate to the" << numberOfWorkingEmployeesOnProject << "employees the" << numberOfTasksToBeDone << "tasks required";
+}
 
-
+float AudiovisualCompany::sellProductsToCustomers(float softwarePrice, int softwarePiecesSold, int numberOfSubscribersToCloudSolutions, float subscriptionPrice)
+{
+    float softwarePiecesFloat;
+    softwarePiecesFloat = float(softwarePiecesSold);
+    float cloudSubscribersFLoat;
+    cloudSubscribersFLoat = float(numberOfSubscribersToCloudSolutions);
```

Comment:
Casting has not been covered, and this is a C-Style cast.
remove it
find another way to solve the warnings related to this type conversion.
- [ ] revise